src/ResourceManagement/AppService/FunctionAppImpl.cs

```diff
@@ -363,7 +363,7 @@ public async Task<string> GetMasterKeyAsync(CancellationToken cancellationToken
                 ServiceClientTracing.Enter(_invocationId, this, "Get", tracingParameters);
             }
             // Construct URL
-            var _baseUrl = string.Format("http://{0}", DefaultHostName().Replace("http://", "").Replace("https://", ""));
```

Comment:
This changes the original implementation details by switch using https to https endpoint? Why original use http for https and what is the reason to change?

Reply:
It uses kudu. "http" will not work if function setting with httpsOnly=true. "https" will work for either httpsOnly=false or httpsOnly=true

Reply:
If this case, if httpsOnly=false, it will always use https, which change the meaning of API. Need conditional based on the httpsOnly settings.

Reply:
"https" should work for both. Kudu would suggest https even if you access http when httpsOnly=false.

```
> curl http://fa3-weidxu.scm.azurewebsites.net/
<head><title>Document Moved</title></head>
<body><h1>Object Moved</h1>This document may be found <a HREF="https://fa3-weidxu.scm.azurewebsites.net/">here</a></body>
```

I can do switch based on httpsOnly config, but since this request actually have Auth token in it, it would be better in HTTPS for all.